bouncer/src/repo/core/model/bson/repo_bson_factory.h

```diff
@@ -152,14 +152,16 @@ namespace repo {
 				*/
 				static RepoUnityAssets makeRepoUnityAssets(
 					const repo::lib::RepoUUID                              &revisionID,
-					const std::vector<std::string>                         &assets,
+					const std::vector<std::string>                         &unityAssetFiles,
```

Comment:
Is this a temporary measure (i.e. will be split when we move to v5 endpoint)? I think we should keep the 2 mapping separate asset mapping separate, considering they have different json and binary files they don't really share anything in common.

Reply:
It is not intended to be - the idea is that having one unityAssets.json, the viewer doesn't need to know in advance which type of bundles it should be asking for.

The other option would for bouncer to write two separate documents, and for .io to return the legacy one if the new one doesn't exist.
This would in some ways be neater because we could add a new API, and the existing one is left for forced-legacy mode.
In this case the viewer would rely on both documents having the same schema, and on the file extension to figure out which type of bundle it should request.

If you'd like me to refactor it to operate like that? Or something else if you have another idea!

Reply:
> This would in some ways be neater because we could add a new API, and the existing one is left for forced-legacy mode.

I think that' would be a neater approach 👍

Reply:
Hi @carmenfan, I am thinking about how to support the two assets lists going forward. 
.io works by looking for the first document that matches the revision id in `stash.unity3d`.

The simplest approach is to make a new collection `stash.repobundles`, and then fallback to `stash.unity3d` if no documents are found. The downside is this adds another collection, and we need to explicitly check the old one in the fallback case indefinitely. 
It is by far the simplest case however and easy to understand.

For the future, it would be nice to have them in one collection with the same access mechanism, because then findOne will simply return the RepoBundles one for new imports and the Unity one for existing imports.
Though, for the period where we have both they need to be disambiguated/need different ids.

So far the only way I can think to do this is to take a similar approach for the `supermeshes_unity.json` fallback.

That is, have `AssetModelExport` start writing the Unity Assets document with a post-fixed name or some member to disambiguate it.

.io will `findOne` against `.unity3d`, returning the RepoBundle one where available, or the Unity one for existing imports. If the unity one is explicitly requested .io will search for that, then after a release hopefully, we simply stop writing the unity one.

Which approach would you prefer? (Or, do you have a better idea!)

Reply:
could we do it so that they're completely separate endpoints, stored in different place. and let unity control the fall back?

i.e. the unity viewer asks for repo bundles asset list -> if response is 404, ask for the asset bundle list?

I'd prefer we don't blur the lines and do the fallback on 3drepo.io and make assumptions on what the user is asking for. especially if the schema might diverge in the future.

3drepo.io as a provider of data I think we want to be explicit, I give you exactly what you're asking for or nothing at all. Whether the client can support a backup option or not is the client's consideration. For e.g. if we moved unreal to use repo bundles or 3rd party integrators using this endpoint, they may not be able to interpret the unity bundles so we shouldn't confuse them

Reply:
@carmenfan yes sounds good! How do you want to handle the supermeshes.json (cache)? Should this be an exception or shall we create a permanent new API (e.g. `supermeshes_repo.json`)?

Reply:
Though what should we do about the assets list for federations that might mix new and existing imports? In this case .io would need to be aware of both because it only returns one response...

Reply:
OK, as discussed, bouncer now writes the assets lists separately to `.stash.unity3d` and to `.stash.repobundles`. The v4 APIs implement backwards compatibility on .io for now, which will be decommissioned when the viewer moves to v5 style APIs and takes over backwards compatibility on a per-container basis.